### PR TITLE
docs: Add How-to guides section with the first guide

### LINKED ISSUE #7854: docs: Add How-to-guides section to APISIX Website API Gateway
### Current State

Currently, we have categorization based on plugins but there is no separation by usability cases.

Actually, plugins cover many things but It is hard to navigate through an exact fit plugin for some use cases without opening the individual plugin tutorial.

For newbies will be quite challenging to understand the core use cases of API Gateway.

### Desired State

We should have a How-to-guides section with learning content per each API Gateway use case that includes the following subpages:

Expose APIs
Protect APIs
Secure API
Monitor APIs
Improve performance. 
Health Check.
Add multiple versions. 
Cache API responses. 
Mock API responses.
Canary release. 
.... more

We have already enough blog posts that cover many API Gateway use cases we can rewrite them for each how-to guide page.

Here is the link to issue #6461

https://github.com/apache/apisix/issues/6461

You can see examples for Azure API Management or Kong API Gateway how the content is structured:

image

image

### PR DESCRIPTION
**Pull request by Boburmirzo**
### Description

Added How-To guide section with the first guide for Observe your APIs which shows use cases of observability plugins. 

Fixes #7854 

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/en/latest/how-to-guides/observe-your-api.md
Comment (hf400159) [inline docs/en/latest/how-to-guides/observe-your-api.md:11]: please add ASF 2.0 license. please refer to: https://github.com/apache/apisix/edit/master/docs/en/latest/plugins/api-breaker.md

Comment (hf400159) [reply docs/en/latest/how-to-guides/observe-your-api.md:11]: image

**Review by hf400159 — COMMENTED** [review]
great! 👍

File: docs/en/latest/how-to-guides/observe-your-api.md
Comment (Boburmirzo) [reply docs/en/latest/how-to-guides/observe-your-api.md:11]: @hf400159 Right, I added now license description.

**Comment by Boburmirzo** [comment]
@juzhiyuan We need one more reviewer and approver:)

File: docs/en/latest/how-to-guides/observe-your-api.md
Comment (SylviaBABY) [inline docs/en/latest/how-to-guides/observe-your-api.md:89]:
```suggestion
```shell
```

File: docs/en/latest/how-to-guides/observe-your-api.md
Comment (SylviaBABY) [inline docs/en/latest/how-to-guides/observe-your-api.md:114]:
```suggestion
```shell
```

File: docs/en/latest/how-to-guides/observe-your-api.md
Comment (SylviaBABY) [inline docs/en/latest/how-to-guides/observe-your-api.md:132]:
```suggestion
```shell
```

File: docs/en/latest/how-to-guides/observe-your-api.md
Comment (SylviaBABY) [inline docs/en/latest/how-to-guides/observe-your-api.md:145]:
```suggestion
```shell
```

File: docs/en/latest/how-to-guides/observe-your-api.md
Comment (SylviaBABY) [inline docs/en/latest/how-to-guides/observe-your-api.md:200]:
```suggestion
```shell
```

File: docs/en/latest/how-to-guides/observe-your-api.md
Comment (SylviaBABY) [inline docs/en/latest/how-to-guides/observe-your-api.md:219]:
```suggestion
```shell
```

**Comment by Boburmirzo** [comment]
@SylviaBABY Thanks, I updated the requested changes.

**Review by SylviaBABY — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Comment by Boburmirzo** [comment]
@hf400159 One question, do these changes go with the next APISIX release, right?

**Comment by hf400159** [comment]
Yes. You can find it here. https://apisix.apache.org/docs/apisix/next/how-to-guides/observe-your-api/   cc@Boburmirzo

**Comment by Boburmirzo** [comment]
> Yes. You can find it here. https://apisix.apache.org/docs/apisix/next/how-to-guides/observe-your-api/ cc@Boburmirzo image

Great! I will work on yet another guide soon.

**Pull request merged**
